src/pages/portfolio/ui/assets-table.tsx

```diff
@@ -0,0 +1,450 @@
+      return parameters?.chainId;
+    },
+    enabled: connectionStore.connected,
+  });
+};
+
+const LoadingState = () => {
+  return (
+    <Card>
+      <div className='p-3'>
+        <Text as={'h4'} large color='text.primary'>
+          Assets
+        </Text>
+
+        {/* Asset distribution bar skeleton */}
+        <div className='w-full h-2 my-5'>
+          <Skeleton />
+        </div>
+
+        {/* Legend skeleton */}
+        <div className='flex flex-wrap gap-4 mb-6'>
+          {Array.from({ length: 6 }).map((_, i) => (
+            <div key={i} className='flex items-center gap-2'>
+              <div className='w-2 h-2'>
+                <Skeleton />
+              </div>
+              <div className='w-20 h-4'>
+                <Skeleton />
+              </div>
+            </div>
+          ))}
+        </div>
+
+        <Density compact>
+          <Table>
+            <Table.Thead>
+              <Table.Tr>
+                <Table.Th>Asset</Table.Th>
+                <Table.Th>Balance</Table.Th>
+                <Table.Th>Price</Table.Th>
+                <Table.Th>Value</Table.Th>
+                <Table.Th>Actions</Table.Th>
+              </Table.Tr>
+            </Table.Thead>
+            <Table.Tbody>
+              {Array.from({ length: 6 }).map((_, i) => (
+                <Table.Tr key={i}>
+                  <Table.Td>
+                    <div className='flex items-center gap-2'>
+                      <div className='w-6 h-6 rounded-full overflow-hidden'>
+                        <Skeleton />
+                      </div>
+                      <div className='w-20 h-5'>
+                        <Skeleton />
+                      </div>
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='w-24 h-5'>
+                      <Skeleton />
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='w-24 h-5'>
+                      <Skeleton />
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='w-24 h-5'>
+                      <Skeleton />
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='flex gap-2'>
+                      <div className='w-8 h-8 rounded-full overflow-hidden'>
+                        <Skeleton />
+                      </div>
+                      <div className='w-8 h-8 rounded-full overflow-hidden'>
+                        <Skeleton />
+                      </div>
+                    </div>
+                  </Table.Td>
+                </Table.Tr>
+              ))}
+            </Table.Tbody>
+          </Table>
+        </Density>
+      </div>
+    </Card>
+  );
+};
+
+const NotConnectedNotice = () => {
+  return (
+    <div className='p-6'>
```

Comment:
chore (ui): this block is smaller than the main portfolio section, please make them the same size to avoid content layout shifts